volume/ceph/ceph.go

```diff
@@ -177,24 +186,64 @@ func (v *Volume) Mount(id string) (mappedDevicePath string, returnedError error)
 	if err != nil {
 		return "", err
 	}
+
+	mappedDevicePath = v.mappedDevicePath
+
+	if fsType == cryptoLUKS {
+		cmd = exec.Command("cryptsetup", "luksOpen", "--key-file=-", v.mappedDevicePath, v.Name())
+		stdin, err := cmd.StdinPipe()
+		if err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+		defer stdin.Close()
+		if err := cmd.Start(); err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		key, err := getLuksKey(v.Name())
+		if err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		io.WriteString(stdin, key)
+		stdin.Close()
+		if err := cmd.Wait(); err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		v.mappedLuksDevicePath = path.Join(LuksDevMapperPath, v.Name())
+		mappedDevicePath = v.mappedLuksDevicePath
+
+		fsType, err = utils.DeviceHasFilesystem(mappedDevicePath)
+		logrus.Errorf("Checked filesystem in %s: %s", mappedDevicePath, fsType)
+		if err != nil {
+			return "", err
+		}
+	}
+
 	if fsType == "" {
-		cmd = exec.Command("mkfs.ext4", "-m0", "-E", "nodiscard,lazy_itable_init=0,lazy_journal_init=0,packed_meta_blocks=1", v.mappedDevicePath)
-		logrus.Infof("Creating ext4 filesystem in newly created Ceph volume '%s' (device %s)", v.Name(), v.mappedDevicePath)
+		//if fsType == "" {
```

Comment:
Fixed in https://github.com/medallia/docker/pull/15/commits/975c9c2ab2e543bd69892b3254de4e3e90c1156e